components/inputs/input-percent.js

```diff
@@ -0,0 +1,180 @@
+ * @slot after - Slot beside the input on the right side. Useful for an "icon" or "button-icon".
+ * @fires change - Dispatched when an alteration to the value is committed (typically after focus is lost) by the user
+ */
+class InputPercent extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(RtlMixin(LitElement)))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Specifies which types of values can be autofilled by the browser
+			 */
+			autocomplete: { type: String },
+			/**
+			 * When set, will automatically place focus on the input
+			 */
+			autofocus: { type: Boolean },
+			/**
+			 * Disables the input
+			 */
+			disabled: { type: Boolean },
+			/**
+			 * Restricts the maximum width of the input box without impacting the width of the label.
+			 */
+			inputWidth: { attribute: 'input-width', type: String },
+			/**
+			 * REQUIRED: Label for the input
+			 */
+			label: { type: String },
+			/**
+			 * Hides the label visually (moves it to the input's "aria-label" attribute)
+			 */
+			labelHidden: { type: Boolean, attribute: 'label-hidden' },
+			/**
+			 * Maximum number of decimal values to show (rounds value up or down).
+			 */
+			maxFractionDigits: { type: Number, attribute: 'max-fraction-digits' },
+			/**
+			 * Minimum number of decimal values to show.
+			 */
+			minFractionDigits: { type: Number, attribute: 'min-fraction-digits' },
+			/**
+			 * Placeholder text
+			 */
+			placeholder: { type: String },
+			/**
+			 * Indicates that a value is required
+			 */
+			required: { type: Boolean },
+			/**
+			 * Text for additional screenreader and mouseover context
+			 */
+			title: { type: String },
+			/**
+			 * Value of the input
+			 */
+			value: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [ super.styles,
+			css`
+				:host {
+					--d2l-input-text-align: end;
+					display: inline-block;
+					position: relative;
+					width: 100%;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				:host([disabled]) [slot="right"] {
+					opacity: 0.5;
+				}
+				[slot="right"] {
+					box-sizing: border-box;
+					cursor: default;
+					padding-left: 0.2rem;
+					padding-right: 0.55rem;
+				}
+				:host([dir="rtl"]) [slot="right"] {
+					padding-left: 0.55rem;
+					padding-right: 0.2rem;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.autofocus = false;
+		this.disabled = false;
+		this.labelHidden = false;
+		this.required = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.label) {
+			console.warn('d2l-input-percent component requires label text');
+		}
+	}
+
+	render() {
+		return html`
+			<d2l-input-number
+				autocomplete="${ifDefined(this.autocomplete)}"
+				?autofocus="${this.autofocus}"
+				@blur="${this._handleBlur}"
+				@change="${this._handleChange}"
+				?disabled="${this.disabled}"
+				.forceInvalid="${this.invalid}"
+				hide-invalid-icon
```

Comment:
Hide the invalid icon because it obscures too much of the value.

Reply:
Another option would be to just never let this go into an invalid state? Like you could always round up to 0 and down to 100 and if they enter something that isn't a number.

Reply:
Yeah I think that could work.  Then I can enable the invalid icon for the required/empty state, which is a nice feature.

Reply:
Got this working, so not for required fields, it will show the icon.
![image](https://user-images.githubusercontent.com/9042472/109198137-d85b4c00-776b-11eb-8114-13dedfd298f9.png)